### ISSUE TITLE
Interactive (manual) channel hopping in airodump-ng

### ISSUE THREAD

**Issue by drws** [aircrack-ng/aircrack-ng#2285]
# Description

This is a feature request for interactive (manual) channel switching in `airodump-ng`'s interface. The idea is that when a keystroke (let's take `c` for example, not already used) is pressed, `airodump-ng` stops auto-hopping and starts to manually increment the currently-active channel every time `c` is pressed (using the channel list, passed with `-c`). Also a manual-off/auto-on hotkey (`x` for example) reverts the default auto-hopping behaviour.

There is also a case when only a single channel is passed to `airodump-ng` with the `-c` argument. In this case the most simple solution would be for the `c` and `x` hotkeys to not have an effect, since auto-hopping algorithm is also not running. However, there is another option that could prove useful, to allow the `c` hotkey to scroll through all possible channels in this case (1-11 or 1-13 in the 2.4GHz band for example). The `x` hotkey could then provide an option to hop back to the default channel (set by `-c`).

**Comment by ChillVibesMushroom** [comment]
So all we have to do to auto hop is not pass the c parameter

**Comment by drws** [comment]
This proposal is specifically about **manual** hopping and switching between manual and auto modes. Auto hopping is already working, but is useless in many real-world scenarios. It's really only good for long-term headless capturing of statistical data.

**Comment by ChillVibesMushroom** [comment]
I did figure out how to generate traffic when attacking a station with multiple frequencies that hop automatically inbetween each other when a target is disconnected and this makes aircrcack-ng that much more beautiful to use

**Comment by drws** [comment]
Yeah, auto-hopping is great for surveying channel-adaptive and/or multi-band networks too, but again, the issue here is that there is currently no manual-hopping. There's single-channel operation and auto-hopping through a predefined list, but no manual-hopping.
